src/sharedHooks/useApiToken.js

```diff
@@ -1,33 +1,32 @@
 // @flow
 
-import { getJWTToken } from "components/LoginSignUp/AuthenticationService";
+import { getJWT } from "components/LoginSignUp/AuthenticationService";
 import { useEffect, useState } from "react";
 import useCurrentUser from "sharedHooks/useCurrentUser";
 
 const useApiToken = ( ): string | null => {
   const [apiToken, setApiToken] = useState( null );
-  const [shouldFetchToken, setShouldFetchToken] = useState( true );
   const currentUser = useCurrentUser( );
 
+  // Without the dependency array, this hook will run on every render, then every time
+  // a new token is fetched, and if expired fetched from API (in getJWT).
```

Comment:
If we call `getJWT()` on ever render, why use an effect at all?

Reply:
Good question. When I made the change I was more in the mindset of how can I reduce what is already there to a point where it works as I want, and did not think about how where that point is when I would build it from the ground up if that makes sense. But I have no preference over either of those two solutions, should I change it to not use a `useEffect`. I am not sure whether there are any technical aspects of that decision that I don't know about yet.

Reply:
Try it without the effect and see if it works. If it does, I'm not really sure useApiToken is necessary at all (I may have added because hooks seemed like the more React-y way of doing work like this, but maybe it was overkill in this case).